CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 ## 0.16.17 (unreleased)
 
+- Fix an issue where the Pulumi CLI would load the newest plugin for a resource provider instead of the version that was
+  requested, which could result in the Pulumi CLI loading a resource provider plugin that is incompatible with the
+  program. This has the potential to disrupt users that previously had working configurations; if you are experiencing
+  problems after upgrading to 0.16.7, you can opt-in to the legacy plugin load behavior by setting the environnment
+  variable `PULUMI_ENABLE_LEGACY_PLUGIN_SEARCH=1`. You can also install plugins that are missing with the command
+  `pulumi plugin install resource <name> <version> --exact`.
```

Comment:
In what case would `--exact` be required?  I think that you are saying if I have 0.16.5 of a plugin installed, and I run `pulumi plugin install 0.16.4` that would be a no-op (since the implicit range here is `^0.16.4`, which `0.16.5` satisfies) but I am wondering how I would ever run into problems from another part of the system?

Reply:
We talked about this in person, but I would recommend us not adding `--exact` and having that be the new default behavior.   When `PULUMI_ENABLE_LEGACY_PLUGIN_SEARCH` is truthy, then `pulumi plugin install resource ...` will behave as it had before this change.  I think that makes installing a plugin and loading a plugin behave in the same manner.

Reply:
That's a bigger change - do you want that as part of this PR or a follow-up?

Reply:
Talked offline, I'll tackle this in the next commit. The idea is to make `pulumi plugin install` behave the same as the plugin loader, respecting the environment variable when deciding what to do.

Reply:
What I think would be good is if `PULUMI_ENABLE_LEGACY_PLUGIN_SEARCH` controlled `pulumi plugin install`'s behavior as well.  When `true` it should behave as if `--exact` was not passed, and when unset, it should behave as if it was.

As a note: It looks like there are no changes to the CLI as part of this PR, did you forget to `get add` something?

Reply:
> As a note: It looks like there are no changes to the CLI as part of this PR, did you forget to `get add` something?

No - the `--exact` flag already exists, I didn't have to change any of that.